src/core/or/connection_edge.c

```diff
@@ -1553,6 +1553,95 @@ consider_plaintext_ports(entry_connection_t *conn, uint16_t port)
   return 0;
 }
 
+/** If address is of the form "y.onion" with a well-formed handle y:
```

Comment:
Hm.I think this func doc can be improved.
 
- Before going into all the various cases let's document what this function is doing in general. Something simple like "Parse an address and set 'type_out' and return blabla"
- An undocumented behavior of this function is that it always sets 'type_out' regardless of success/fail. This is a behavior that other code seems to rely on and we should document it.

Reply:
Very good catch.

Fixup commit: 6a730b0118